tests/unit/hive_metastore_client/test_hive_mestastore_client.py

```diff
@@ -93,6 +93,31 @@ def test_add_columns_to_table(
             dbname=db_name, tbl_name=table_name, new_tbl=mocked_return_get_table
         )
 
+    @mock.patch.object(HiveMetastoreClient, "get_table")
+    @mock.patch.object(HiveMetastoreClient, "alter_table")
+    def test_drop_columns_from_table(
+        self, mocked_alter_table, mocked_get_table, hive_metastore_client
+    ):
+        # arrange
+        db_name = "db_name"
+        table_name = "table_name"
+        cols = ["col1", "col2"]
+
+        mocked_return_get_table = Mock()
+        mocked_return_get_table.sd.cols = []
+        mocked_get_table.return_value = mocked_return_get_table
```

Comment:
in this test the get` `table returns a table with no columns, then you try to delete the cols ["col1", "col2"]. This test could be improved.

I think this test could have a parameterize that tests the callings of `mocked_alter_table` with the cols list. To test your for (this test is missing).

About the for looping (from lines 100 to 104) testing: if you think that it applies you can move the for into a separate function and test it separately too.